### PR TITLE
Require Login for group and event creation page

### PR DESCRIPTION
**Pull request by eiffelwong1**
## Summary (1-2 sentences)

require login for pages that would need access control in the future

## Details (reason and description of the changes)

requires login in New group page and New event page

### CODE REVIEW

File: app/modules/session/session.server.ts [hunk 1]
```diff
@@ -69,3 +69,11 @@ export async function logout(request: Request) {
     },
   });
 }
+
+export async function requireUserSession(request: Request) {
+  const session = await getUserSession(request);
+  if (!session) {
+    return redirect('/login');
```
Comment (andrelandgraf) [inline app/modules/session/session.server.ts:76]:
```suggestion
    throw redirect('/login');
```

**Review by andrelandgraf — COMMENTED** [review]
One comment, rest looks great! Thanks!

File: app/modules/session/session.server.ts (hunk 1 above)
Comment (andrelandgraf) [reply app/modules/session/session.server.ts:76]: Context:
- https://sergiodxa.com/articles/throwing-vs-returning-responses-in-remix
- https://remix.run/docs/en/main/guides/errors

**Review by tonydangblog — APPROVED** [review]
This is a pretty useful. Not having to check that `session` exists every time saves a lot of typing :)

**Review by tonydangblog — APPROVED** [review]
LGTM

File: app/routes/groups_.$groupId_.events_.new.tsx [hunk 2]
```diff
@@ -7,6 +7,11 @@ import { Card } from '~/components/ui/containers';
 import { Input, TextArea } from '~/components/ui/forms';
 import { Button } from '~/components/ui/button';
 import { H1 } from '~/components/ui/headers';
+import { requireUserSession } from '~/modules/session/session.server';
+
+export async function loader({ request }: LoaderFunctionArgs) {
```
Comment (andrelandgraf) [inline app/routes/groups_.$groupId_.events_.new.tsx:12]: As a next step, we could also check here that the user has "admin" access to the group before we allow them to create a new event for that group!

**Pull request merged**